src/App.js

```diff
@@ -1,85 +1,129 @@
+  const [twitter, setTwitter] = useState('')
+  const [name, setName] = useState('')
+  const [profile, setProfile] = useState({})
+  const [localDid, setDid] = useState(null)
+  const [idxInstance, setIdxInstance] = useState(null)
+  const [loaded, setLoaded] = useState(false)
+  const [showGreeting, setShowGreeting] = useState(false)
+  const idxRef = useRef(null)
+  const didRef = useRef(null)
+  idxRef.current = idxInstance
+  didRef.current = localDid
 
-  return (
-    <>
-      <section className="text-gray-600 body-font bg-gray-50">
-        <div className="container mx-auto flex px-5 py-24 items-center justify-center flex-col">
-          <img
-            src={logo}
-            alt="hero"
-            className="mb-5 w-32 h-32 object-cover object-center rounded-full"
-          />
-
-          <div className="text-center lg:w-2/3 w-full">
-            <div className="flex justify-center">
-              <div className="relative mr-4 lg:w-full xl:w-1/2 w-2/4 md:w-full text-left">
-                <h1 className="mb-6 text-xl text-center">
-                  Search developer DAO id
-                </h1>
-                <input
-                  placeholder="Search developer id"
-                  value={developerId}
-                  onChange={(e) => updateDeveloperId(e.target.value)}
-                  type="text"
-                  id="hero-field"
-                  name="hero-field"
-                  className="w-full mb-6 bg-white bg-opacity-50 rounded focus:ring-2 focus:ring-indigo-200 focus:bg-transparent border border-gray-300 focus:border-indigo-500 text-base outline-none text-gray-700 py-1 px-3 leading-8 transition-colors duration-200 ease-in-out"
-                />
-              </div>
-            </div>
-          </div>
-          <NftProvider fetcher={["ethers", ethersConfig]}>
-            <Nft developerId={developerId} />
-          </NftProvider>
-        </div>
-      </section>
-    </>
-  );
-}
+  async function connect() {
+    const cdata = await client()
+    const { did, idx, error } = cdata
+    if (error) {
+      console.log('error: ', error)
+      return
+    }
+    setDid(did)
+    setIdxInstance(idx)
+    const data = await idx.get('basicProfile', did.id)
+    if (data) {
+      setProfile(data)
+    } else {
+      setShowGreeting(true)
+    }
+    setLoaded(true)
+  }
 
-function Nft(developerId) {
-  const { loading, error, nft } = useNft(
-    "0x25ed58c027921E14D86380eA2646E3a1B5C55A8b",
-    developerId.developerId
-  );
+  async function updateProfile() {
+    if (!twitter && !bio && !name) {
+      console.log('error... no profile information submitted')
+      return
+    }
+    if (!idxInstance) {
+      await connect()
+    }
+    const user = {...profile}
+    if (twitter) user.twitter = twitter
+    if (bio) user.bio = bio
+    if (name) user.name = name
+    await idxRef.current.set('basicProfile', user)
+    setLocalProfileData()
+    console.log('profile updated...')
```

Comment:
```suggestion
    alert('profile updated...')
```

A more visual confirmation would be nice! Even a simple `alert()` would do for now.

Reply:
I'd recommend a [Toast](https://chakra-ui.com/docs/feedback/toast) to one-up alerts, even, now that we're using Chakra-UI.